k8s/oidc-vault/README.md

```diff
@@ -0,0 +1,458 @@
+
+```console
+$ kubectl get service -n spire spire-oidc
+
+NAME           TYPE           CLUSTER-IP    EXTERNAL-IP    PORT(S)          AGE
+spire-oidc     LoadBalancer   10.12.0.18    34.82.139.13   443:30198/TCP    108s
+```
+
+## Configure an A Record for the OIDC Discovery Document Endpoint
+
+Using your preferred DNS tool, put the `MY_DISCOVERY_DOMAIN` domain and the `spire-oidc` external IP address in a new DNS A record. The A record should take the following form:
+
+```console
+MY_DISCOVERY_DOMAIN          A        <EXTERNAL-IP for spire-oidc service>
+```
+
+For example:
+```console
+oidc-discovery.example.org   A        34.82.139.13
+```
+
+Note: Do not use the `oidc-discovery.example.org` domain or IP address shown above.
+
+## Verify the DNS A Record
+
+As with any change to DNS, it will take minutes or hours for the new A record to propagate to DNS servers. This tutorial will not work until the A record is propagated. Negative DNS query results will be cached, causing headaches. So, to be safe, wait an hour or so to test the DNS change after creating the A record.
+
+1. Use `nslookup` to display the DNS information for the domain you configured in the A record:
+
+   ```console
+   $ nslookup oidc-discovery.example.org
+   Server:        203.0.113.0
+   Address:	      203.0.113.0#53
+
+   Non-authoritative answer:
+   Name:	oidc-discovery.example.org
+   Address: 93.184.216.34
+   ```
+
+   The `Address:` field at the bottom should correspond to the IP address in the A record.
+
+2. In your browser, navigate to `https://MY_DISCOVERY_DOMAIN/.well-known/openid-configuration`. You should see a JSON output similar to the following:
+
+   ```json
+   {
+     "issuer": "https://oidc-discovery.example.org",
+     "jwks_uri": "https://oidc-discovery.example.org/keys",
+     "authorization_endpoint": "",
+     "response_types_supported": [
+       "id_token"
+     ],
+     "subject_types_supported": [],
+     "id_token_signing_alg_values_supported": [
+       "RS256",
+       "ES256",
+       "ES384"
+     ]
+   }
+   ```
+
+
+# Part 3: Install and Configure the Vault Server
+
+After configuring an A Record for the OIDC Discovery document endpoint, continue with configuring the Vault server.
+
+## Install Vault
+
+Before proceeding, ensure that you have installed HashiCorp Vault on your local computer and that the location of the `vault` executable is in your system `PATH`. You can verify that Vault is installed by typing `vault version` in a terminal window, which outputs the Vault version if it is already installed, or a _command not found_ error otherwise.
+
+If the command fails you need to install Vault. There are a few install methods depending on your operating system, like packages for some Linux flavors, Homebrew (for macOS) and Chocolatey (for Windows). You can also [download a precompiled binary](https://www.vaultproject.io/downloads) for your operating system.
```

Comment:
not sure if this is what you mean:

```markdown
Before proceeding, ensure that you have installed HashiCorp Vault on your local computer and that the location of the `vault` executable is in your system `PATH`. You can verify that Vault is installed by typing `vault version` in a terminal window, which outputs the Vault version if it is already installed, or a _command not found_ error otherwise.

If the command fails, you need to install Vault. There are a few install methods depending on your operating system, like packages for some Linux flavors, Homebrew (for macOS), and Chocolatey (for Windows). You can also [download a precompiled binary](https://www.vaultproject.io/downloads) for your operating system. For a complete list of installation instructions, see the [Install Vault](https://www.vaultproject.io/docs/install) section in the documentation site, or the [Getting Started - Install Vault](https://learn.hashicorp.com/tutorials/vault/getting-started-install) guide.
 
Use a new Vault installation for this tutorial. An existing Vault installation may have configuration settings that conflict with this tutorial. 
```

I concatenated the two paragraphs and added the third one, but it feels like the later should be in the second paragraph too.

Reply:
Hmm, yeah, this is all Vault installation info. Having one big paragraph instead of three makes it look like a textbook instead of easily digestible info. So we need some logical division of these groups of sentences. I do think the fresh install info is slightly different enough to be on its own. I edited the main install info so it fits into one paragraph. Here's a new version with everything:

-----
Ensure that HashiCorp Vault is installed on your local computer and that the location of the `vault` executable is in your system `PATH`. Verify that Vault is installed by typing `vault version` in a terminal window. The output indicates Vault version if it is installed, or a _command not found_ error otherwise, in which case you need to install Vault. Packages are available for some Linux flavors, MacOS (via Homebrew) and Windows (via Chocolatey). Alternatively, you can [download a precompiled binary](https://www.vaultproject.io/downloads) for your operating system. For complete installation instructions, see the [Install Vault](https://www.vaultproject.io/docs/install) or the [Getting Started - Install Vault](https://learn.hashicorp.com/tutorials/vault/getting-started-install).
 
Use a new Vault installation for this tutorial. An existing Vault installation may have configuration settings that conflict with this tutorial.

Reply:
you rocks 🚀